lib/reporters/base.js

```diff
@@ -156,14 +156,8 @@ exports.cursor = {
 };
 
 function showDiff (err) {
-  return err && err.showDiff !== false && sameType(err.actual, err.expected) && err.expected !== undefined;
-}
-
-function stringifyDiffObjs (err) {
-  if (!utils.isString(err.actual) || !utils.isString(err.expected)) {
-    err.actual = utils.stringify(err.actual);
-    err.expected = utils.stringify(err.expected);
-  }
+  return (utils.type(err.diff) === 'function') ||
+         (err.showDiff !== false && sameType(err.actual, err.expected) && err.expected !== undefined);
```

Comment:
please use `typeof err.expected === 'undefined'` for safety

Reply:
Ok. Maybe use `utils.type(err.expected) === 'undefined'` for uniformity with a code in the line above?